cloudprovider/aliyun/aliyun_provider_test.go

```diff
@@ -0,0 +1,70 @@
+package aliyun
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"gopkg.in/check.v1"
+
+	"github.com/rancher/agent/cloudprovider"
+	"github.com/rancher/agent/core/hostInfo"
+	"github.com/rancher/agent/utilities/config"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) {
+	check.TestingT(t)
+}
+
+type ComputeTestSuite struct {
+}
+
+var _ = check.Suite(&ComputeTestSuite{})
+
+func (s *ComputeTestSuite) SetUpSuite(c *check.C) {
+}
+
+type fakeReplyImpl struct{}
+
+func (f fakeReplyImpl) Region() (string, error) {
+	return "fake", nil
+}
+
+func (f fakeReplyImpl) Zone() (string, error) {
+	return "fake", nil
+}
+
+type errorReplyImpl struct{}
+
+func (e errorReplyImpl) Region() (string, error) {
+	return "", errors.New("fake error")
+}
+
+func (e errorReplyImpl) Zone() (string, error) {
+	return "", errors.New("fake error")
+}
+
+func (s *ComputeTestSuite) TestGetHostInfo(c *check.C) {
+	os.Mkdir(config.StateDir(), 0755)
+	p := Provider{
+		interval:   time.Second,
+		expireTime: time.Second * 5,
+	}
+
+	i := hostInfo.Info{}
+	i.Labels = map[string]string{
+		cloudprovider.RegionLabel:           "fake",
+		cloudprovider.AvailabilityZoneLabel: "fake",
+		cloudprovider.CloudProviderLabel:    aliyunTag,
+	}
+
+	p.client = fakeReplyImpl{}
+	hostInfo, err := p.GetHostInfo()
+	c.Assert(hostInfo, check.DeepEquals, i)
+
+	p.client = errorReplyImpl{}
+	hostInfo, err = p.GetHostInfo()
+	c.Assert(err.Error(), check.Equals, "fake error")
```

Comment:
Use `ErrorMatches`, see https://godoc.org/gopkg.in/check.v1#Checker

Reply:
it could not just do simple way like this c.Assert(err, Matchs, "fake*"), the type error don't have String() methods, and return "Obtained value is not a string and has no .String()", so I choose to use simple equal.
https://github.com/matttproud/gocheck/blob/master/checkers.go

Reply:
The checker `ErrorMatches` is for type error.